### PR TITLE
e2e: stop controller if context cancelled

### PR DESCRIPTION
**Pull request by aojea**
/kind cleanup
```release-note
NONE
```

Seen in some test with failures, if the test fails the context cancel but the informers keep retrying leaving a lot of noise and red herring in the tests output

https://prow.k8s.io/view/gs/kubernetes-ci-logs/logs/ci-kubernetes-ec2-arm64-conformance-latest/1849930373073473536

### CODE REVIEW

File: test/e2e/framework/service/jig.go [hunk 1]
```diff
@@ -355,9 +355,10 @@ func (j *TestJig) WaitForEndpointOnNode(ctx context.Context, nodeName string) er
 
 // waitForAvailableEndpoint waits for at least 1 endpoint to be available till timeout
 func (j *TestJig) waitForAvailableEndpoint(ctx context.Context, timeout time.Duration) error {
+	internalCtx, cancel := context.WithTimeout(ctx, timeout)
```
Comment (pohly) [inline test/e2e/framework/service/jig.go:358]: I would replace `ctx`:
```
ctx, cancel := context.WithTimeout(ctx, timeout)
defer cancel()
```

Then you don't need update the rest of the code, except for `wait.PollUntilContextTimeout` which can be replaced by `wait.PollUntilWithContext`.

The advantage is that it becomes impossible to accidentally use the `ctx` without the timeout.

Comment (aojea) [reply test/e2e/framework/service/jig.go:358]: 👍

Comment (aojea) [reply test/e2e/framework/service/jig.go]: done

**Review by pohly — APPROVED** [review]
/lgtm
/approve

**Pull request merged**